packages/dashboards/addon/components/navi-widget.js

```diff
@@ -1,71 +1,96 @@
+ *     @canEdit={{true}}
+ *   />
  */
+import { A as arr } from '@ember/array';
 import Component from '@ember/component';
 import { get, computed } from '@ember/object';
 import layout from '../templates/components/navi-widget';
+import { layout as templateLayout, tagName } from '@ember-decorators/component';
 
-export default Component.extend({
-  layout,
-
-  /**
-   * @property {String} tagName - don't add a wrapper tag for this component
-   */
-  tagName: '',
-
+@templateLayout(layout)
+@tagName('')
+class NaviWidget extends Component {
   /**
    * @property {String} visualizationPrefix - prefix for all visualizations types
    */
-  visualizationPrefix: 'navi-visualizations/',
+  visualizationPrefix = 'navi-visualizations/';
 
   /**
-   * @property {Object} widget - model for widget
+   * @property {Object} model - widget model
    */
-  model: undefined,
+  model = undefined;
 
   /**
-   * @property {DS.PromiseObject} data - data for widget
+   * @property {TaskInstance} taskInstance - data task instance
    */
-  data: undefined,
+  taskInstance = undefined;
 
   /**
    * @property {Object} layoutOptions - layout for dashboard presentation
    */
-  layoutOptions: undefined,
+  layoutOptions = undefined;
 
   /**
    * @property {Object} options - object for grid-stack-item
    */
-  options: computed('layoutOptions.@each.{width,height,row,column,widgetId}', 'model.id', function() {
-    let layout = get(this, 'layoutOptions'),
-      id = get(this, 'model.id');
+  @computed('layoutOptions.@each.{width,height,row,column,widgetId}', 'model.id')
+  get options() {
+    const {
+      layoutOptions: layout,
+      model: { id }
+    } = this;
 
     if (layout) {
       // Map layout to gridstack options
-      let { column: x, row: y, height, width } = layout;
+      const { column: x, row: y, height, width } = layout;
       return { id, x, y, height, width };
     }
-  }),
+
+    return {};
+  }
+
+  /**
+   * @property {Boolean} isRunning - whether widget data is loading
+   */
+  @computed('taskInstance.isRunning')
+  get isRunning() {
+    const { taskInstance } = this;
+
+    return !taskInstance || taskInstance.isRunning;
+  }
 
   /**
    * @property {String} filterErrors - Error messaging for filters that couldn't be applied to the widget
    */
-  filterErrors: computed('data.isFulfilled', function() {
+  @computed('data')
+  get filterErrors() {
     const filterErrors = get(this, 'data.firstObject.response.meta.errors') || [];
     const filterErrorMessages = filterErrors
       .filter(e => e.title === 'Invalid Filter')
       .map(e => e.detail)
       .join('\n');
 
     return filterErrorMessages ? `Unable to apply filter(s):\n${filterErrorMessages}` : null;
-  })
-});
+  }
+
+  /**
+   * @property {EmberArray|Null} data - widget data
+   */
+  @computed('taskInstance.{isSuccessful,value}')
+  get data() {
+    const { isSuccessful, value } = this.taskInstance || {};
+
+    return isSuccessful ? arr(value) : null;
```

Comment:
What is the value of `value` when `isSuccessful` is falsey?

Reply:
`null`